service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java

```diff
@@ -38,6 +40,19 @@
      * @throws IOException This exception is thrown if the object creation failed
      */
     public static ServiceClient createFromConnectionString(String connectionString, IotHubServiceClientProtocol iotHubServiceClientProtocol) throws IOException
+    {
+        return createFromConnectionString(connectionString, iotHubServiceClientProtocol, null);
+    }
+
+    /**
+     * Create ServiceClient from the specified connection string
+     * @param iotHubServiceClientProtocol  protocol to use
+     * @param connectionString The connection string for the IotHub
+     * @param proxyOptions The proxy options to use when connecting to the service. May be null if no proxy will be used.
+     * @return The created ServiceClient object
+     * @throws IOException This exception is thrown if the object creation failed
+     */
+    public static ServiceClient createFromConnectionString(String connectionString, IotHubServiceClientProtocol iotHubServiceClientProtocol, ProxyOptions proxyOptions) throws IOException
```

Comment:
Why not group this under connection options which has protocol, proxyOptions etc and lets you add future connection options?

Reply:
That depends on how long we plan on supporting this particular SDK. I didn't envision bringing any other options here since we will be track 2 soon. But what do you think?

Reply:
I don't forsee either (As it is hard to see for 10 years down the line). I would still prefer to group because we have to support this for 10 years even after we move to track-2 ;just to be on safer side.

Reply:
Do you picture a class "ConnectionOptions" being reused from AMQP to HTTP, though? Or do you picture an HttpConnectionOptions class for registry manager, and an AmqpConnectionOptions class for service client, where each class has just the proxy to set?

Reply:
I pushed a commit that shows one way to have a more general "RegistryManagerOptions" and "ServiceClientOptions" class structure. Can you take a look?